### PR TITLE
refactor(server): asset service - upload asset

### PR DESCRIPTION
**Pull request by jrasm91**
This PR creates an `AssetCore`, where refactored `AssetService`, `AssetController`, and `AssetRepository` logic will eventually migrate to, starting with `uploadFile`/`createAsset`.

TODO:
- [x] Decouple service from `Express.Multer.File`
- [x] Streamline asset upload process (asset vs live photo)
- [x] Remove background service and use `IJobRepository` directly
- [x] Add file upload test for success and duplicate scenario
- [x] Prevent `!isVisible` videos from being processed by the metadata extraction job
- [x] Add file upload test for live photo
- [x] Add delete assets test

### CODE REVIEW

File: server/apps/immich/src/api-v1/asset/dto/create-asset.dto.ts [hunk 1]
```diff
@@ -22,9 +23,29 @@ export class CreateAssetDto {
   @IsNotEmpty()
   isFavorite!: boolean;
 
+  @IsOptional()
```
Comment (alextran1502) [inline server/apps/immich/src/api-v1/asset/dto/create-asset.dto.ts:26]: Do we forget to run OpenAPI generator in this PR?

Comment (jrasm91) [reply server/apps/immich/src/api-v1/asset/dto/create-asset.dto.ts:26]: Looks like I did, yeah.
